server/controllers/authentication.coffee

```diff
@@ -93,14 +93,19 @@ module.exports.saveUnauthenticatedUser = (req, res, next) ->
         User.all (err, users) ->
             return next new Error err if err
             # if existing user document with password -> request rejected
-            if users[0]?.password
+            if users[0]?.password and not requestData.password?
                 error        = new Error 'Not authorized'
                 error.status = 401
                 return next error
             else if users.length
                 users[0].merge userToSave, (err) ->
                     return next new Error err if err
-                    res.status(200).send(result: 'User data correctly updated')
+                    if requestData.password?
+                        return Authentication.authenticate(req, res, next)
```

Comment:
For my culture, why do you use the authenticate function here? This function is about saving unauthenticated informations only when authorized. If I understand correctly, you want to authenticate the user here when his password is saved? I think we should redirect to the login page with the ```next``` argument in order to require user authentication (see my comment below). What do you think about?